app/controllers/questions_controller.rb

```diff
@@ -32,6 +32,30 @@ def create
 
     respond_to do |format|
       if @question.save
+        # check if there are any tags for the question
+        if params[:tags_list]
+          # get names of tags the user wants to add to the question
+          tag_names_array = params[:tags_list].split(",")
+          # tags[] = array for storing Tag instances associated with the names in tag_names_array
+          tags = []
```

Comment:
It looks like you don't need this `tags` array, I think you can just push directly onto `@question.tags` (but you may want to clear it first like `@questions.tags = []`